doc/make_tables.py

```diff
@@ -201,49 +252,48 @@ class ColorTable(DocTable):
     """Class to generate colors table."""
 
     path = f"{COLORS_TABLE_DIR}/colors.rst"
-    header = textwrap.dedent(
+    header = _aligned_dedent(
         """
-        .. list-table::
-           :widths: 50 20 30
-           :header-rows: 1
-
-           * - Name
-             - Hex value
-             - Example
+        |.. list-table::
+        |   :widths: 50 20 30
+        |   :header-rows: 1
+        |
+        |   * - Name
+        |     - Hex value
+        |     - Example
         """
     )
-    row_template = textwrap.indent(
-        textwrap.dedent(
-            """
-            * - {}
-              - ``{}``
-              - .. raw:: html
-
-                   <span style='width:100%; height:100%; display:block; background-color: {};'>&nbsp;</span>
-            """
-        ),
-        "   ",  # Extra indent needed to make it part of the table
+    row_template = _aligned_dedent(
+        """
+        |   * - {}
+        |     - ``{}``
+        |     - .. raw:: html
+        |
+        |          <span style='width:100%; height:100%; display:block; background-color: {};'>&nbsp;</span>
+        """
     )
 
     @classmethod
-    def fetch_tokens(cls):
-        tokens = {token: {"hex": hex} for token, hex in pv.hexcolors.items()}
-        for s, token in pv.colors.color_synonyms.items():
-            if "synonyms" not in tokens[token]:
-                tokens[token]["synonyms"] = []
-            tokens[token]["synonyms"].append(s)
-        return tokens
+    def fetch_data(cls):
+        # Fetch table data from ``hexcolors`` dictionary.
+        colors = {name: {"name": name, "hex": hex} for name, hex in pv.hexcolors.items()}
+        # Add synonyms defined in ``color_synonyms`` dictionary.
+        for s, name in pv.colors.color_synonyms.items():
+            if "synonyms" not in colors[name]:
+                colors[name]["synonyms"] = []
```

Comment:
I don't think I understand the logic here. `colors[name]` is a dict like `{'name': name, 'hex': hex}`, so we know for a fact that `'synonyms' is not in colors[name]`. Am I missing something?

Looking at the functionality, I would expect instead that we just want something like this:
```py
colors = {name: {"name": name, "hex": hex, "synonyms": []} for name, hex in pv.hexcolors.items()}
```

I first wanted to suggest a `defaultdict(list)` but that's probably not necessary.

Reply:
Yea, that's better, will update it.

Reply:
Ah, OK, I get it, it was semantically correct because after the first pass the key was present with the given list.